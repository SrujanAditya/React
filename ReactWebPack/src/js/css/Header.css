.headContainer {
    max-width: 90%;
    display: block;
    margin: auto
}
.searchBox {
    background: black;
    color: white;
    width: 100%;
    line-height: 200%
}
.netflixText {
    display: block;
    color:red;
    font-size: 12px;
    padding: 10px;
    font-weight: bold 
}
.label {
    color:white;
    font-weight: bolder;
    line-height: 300%
}
.searchByText {
    font-size: 11px;
    font-weight: bolder
}
.searchByButtons {
    align-content: center;
    padding:3px;
    font-size: 11px;
    font-weight: 900;
    color: white;
    width: 100px;
    margin: 5px;
    border: none 
}
.searchByButtons:hover {
    box-shadow: 3px 3px 3px 3px rgba(255,255,255,0.3)
}
.searchButton {
    align-content: center;
    padding:5px;
    font-size: 12px;
    font-weight: 900;
    color: white;
    width: 180px;
    background: red;
    border: 1px none;
    float: right;
}
.searchButton:hover {
    box-shadow: 3px 3px 3px 3px rgba(255,255,255,0.3)
}
.mainHeader{
    display: block;
    background: url("../images/background_netflix_2018_by_mrfantasmabr-dbz3o4b.png")
}
