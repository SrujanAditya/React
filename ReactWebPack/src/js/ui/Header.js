import React, {Component} from 'react';
import '../css/Header.css';

class Header extends Component{ 
    constructor(props) {
        super(props);
        this.searchText="";
        this.searchBy="title";
        this.bgColorTitle="red";
        this.bgColorGenre="grey";
    }
    changeSearchByTitle = () => {
        this.bgColorTitle="red";
        this.bgColorGenre="grey";
        this.searchBy="title";
        this.forceUpdate();
    }
    changeSearchByGenre = () => {
        this.bgColorTitle="grey";
        this.bgColorGenre="red";
        this.searchBy="genres";
        this.forceUpdate();
    }
    searchTextChange = () => this.searchText=this.searchValue.value;
    onSubmit = () => this.props.fetchData(this.searchText,this.searchBy);
    checkEnter = (e) => {
        if(e.key === "Enter"){
            this.onSubmit();
        }
    }
    render() {
        return (
            <div className="mainHeader">
                <div className = "headContainer">
                    <label className = "netflixText" >netflixroulette</label>
                    <label className = "label" >FIND YOUR MOVIE</label>
                    <input className = "searchBox" type="text" placeholder=" Search" 
                            onChange = {this.searchTextChange}
                            onKeyPress={this.checkEnter}
                            ref={(input) => this.searchValue = input} />  
                    <div  style={{display:'inline-block',width:'100%',margin:'5px 0px'}}>
                        <div style={{display:'inline'}}>
                            <label className = "label searchByText">SEARCH BY </label>
                            <button id = "titleBtn" className = "searchByButtons" 
                                    onClick={this.changeSearchByTitle}
                                    style={{backgroundColor:this.bgColorTitle}}>TITLE</button>
                            <button id = "genreBtn" className = "searchByButtons" 
                                    onClick = {this.changeSearchByGenre}
                                    style={{backgroundColor:this.bgColorGenre}}>GENRE</button>
                        </div>
                        <div style={{display:'inline'}}>
                            <button className= "searchButton" 
                                    onClick={this.onSubmit}>SEARCH</button>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default Header;