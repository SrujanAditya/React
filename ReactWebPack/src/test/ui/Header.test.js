import React from 'react';
import Enzyme, { shallow , mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Header from '../../js/ui/Header';

Enzyme.configure({ adapter: new Adapter() })

function setup() {
    const props = {
        fetchData : jest.fn()
    }
    const enzymeWrapper = shallow(<Header {...props}/>)

    return {
        props,
        enzymeWrapper
    }
}

describe('ui', () => {
  describe('Header', () => {
    const { enzymeWrapper } = setup();

    enzymeWrapper.find("#titleBtn").simulate("click");

    it('check title button click and color changed to red', () => { 
        expect(enzymeWrapper.instance().bgColorTitle).toEqual("red");
        expect(enzymeWrapper.instance().bgColorGenre).toEqual("grey");
    })
  })
}) 

describe('ui', () => {
    describe('Header', () => {
      const { enzymeWrapper } = setup();
  
      enzymeWrapper.find("#genreBtn").simulate("click");
  
      it('check genre button click and color changed to red', () => { 
          expect(enzymeWrapper.instance().bgColorTitle).toEqual("grey");
          expect(enzymeWrapper.instance().bgColorGenre).toEqual("red");
      })
    })
}) 

describe('ui', () => {
    describe('Header', () => {
       const { props } = setup();
        const wrapper = mount(<Header {...props} />);
      it('check Submit button click', () => { 
        const spy = jest.spyOn(wrapper.instance(), 'onSubmit');
        wrapper.update();
        wrapper.find(".searchButton").simulate('click');
        expect(spy).not.toHaveBeenCalled();
      })
    })
})

describe('ui', () => {
    describe('Header', () => {
       const { props } = setup();
        const wrapper = mount(<Header {...props} />);
      it('check Enter function called on clicking enter key', () => { 
        const spy = jest.spyOn(wrapper.instance(), 'checkEnter');
        wrapper.update();
        wrapper.find(".searchBox").simulate('keypress',{key:'Enter'});
        expect(spy).not.toHaveBeenCalled();
      })
    })
})

describe('ui', () => {
    describe('Header', () => {
       const { props } = setup();
        const wrapper = mount(<Header {...props} />);
      it('check Enter function called on clicking A letter key', () => { 
        const spy = jest.spyOn(wrapper.instance(), 'checkEnter');
        wrapper.update();
        wrapper.find(".searchBox").simulate('keypress',{key:'A'});
        expect(spy).not.toHaveBeenCalled();
      })
    })
})

describe('ui', () => {
    describe('Header', () => {
       const { props } = setup();
        const wrapper = mount(<Header {...props} />);
      it('check searchTextChange function called on text change in search box', () => { 
        const spy = jest.spyOn(wrapper.instance(), 'searchTextChange');
        wrapper.update();
        wrapper.find(".searchBox").simulate('change',{target:{value:'kill'}});
        expect(spy).not.toHaveBeenCalled();
      })
    })
})

describe('ui', () => {
    it('Header component renders correctly', () => {
        const HeaderComp = renderer.create(<Header />).toJSON();
        expect(HeaderComp).toMatchSnapshot();
    }) 
})
