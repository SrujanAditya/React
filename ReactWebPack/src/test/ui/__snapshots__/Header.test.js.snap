// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui Header component renders correctly 1`] = `
<div
  className="mainHeader"
>
  <div
    className="headContainer"
  >
    <label
      className="netflixText"
    >
      netflixroulette
    </label>
    <label
      className="label"
    >
      FIND YOUR MOVIE
    </label>
    <input
      className="searchBox"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder=" Search"
      type="text"
    />
    <div
      style={
        Object {
          "display": "inline-block",
          "margin": "5px 0px",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline",
          }
        }
      >
        <label
          className="label searchByText"
        >
          SEARCH BY 
        </label>
        <button
          className="searchByButtons"
          id="titleBtn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "red",
            }
          }
        >
          TITLE
        </button>
        <button
          className="searchByButtons"
          id="genreBtn"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "grey",
            }
          }
        >
          GENRE
        </button>
      </div>
      <div
        style={
          Object {
            "display": "inline",
          }
        }
      >
        <button
          className="searchButton"
          onClick={[Function]}
        >
          SEARCH
        </button>
      </div>
    </div>
  </div>
</div>
`;
